core/src/com/unciv/logic/map/tile/TileImprovementFunctions.kt

```diff
@@ -316,9 +316,12 @@ class TileImprovementFunctions(val tile: Tile) {
         if (distance > 5) return
         var stats = Stats()
         for (unique in tile.getTerrainMatchingUniques(UniqueType.ProductionBonusWhenRemoved)) {
+            var statsToAdd = unique.stats
             if (unique.isModifiedByGameSpeed())
-                stats.add(unique.stats * civ.gameInfo.speed.modifier)
-            else stats.add(unique.stats)
+                statsToAdd *= civ.gameInfo.speed.modifier
+            if (unique.isModifiedByGameProgress())
+                statsToAdd *= unique.modifyByGameProgress(civ, unique.getModifiers(UniqueType.ModifiedByGameProgress).firstOrNull()!!.params[0].toFloat())
```

Comment:
There's no need for `firstOrNull()!!`, that's equivalent to `first()`
But in general I disagree, if there are multiple modifiers they should **all** modify
I'm not sure if addatively or multiplicatively, I'll leave that to you. In any case this should all be within the getGameProgressModifier function, 'foreach modifier in unique.getModifiers(UniqueType.ModifiedByGameProgress)'

Reply:
Game speed modifier and game progress modifier should of course stack multiplicatively, what do you mean by multiple modifiers? To add multiple modifiers to scale something by gamespeed^2 or gameprogress^2?

Reply:
In general, any unique can have a list of modifiers, and all number-based modifiers should stack. Here you only take the first game progress modifier. What if there are 2? Even, 2 with different numbers? As a modder I would expect them to stack

Reply:
In such a case you can multiplicatively stack them to scale the value with a factor x, x^2, x^3 etc. But instead of trying to get a polynomial fit this way, there should maybe be a unique to rescale the value exponentially, as is the case with for example military units that get approximately 1.413 times stronger each era.

Reply:
??
You're thinking too hard
In the modifyByGameProgress we should just start with 1f, and then, for each modifier, multiply it for that modifier.
That's it :)

Reply:
Ok, I was just a but on confused why a modder would want to scale it with gameprogress squared.